src/integration/data/vectors/upsertAndUpdate.test.ts

```diff
@@ -96,13 +96,14 @@ describe('Testing retry logic on Upsert operation, as run on a mock, in-memory h
   let server: http.Server; // Note: server cannot be something like an express server due to conflicts w/edge runtime
   let mockServerlessIndex: Index;
   let callCount: number;
+  let op: string;
 
   // Helper function to start the server with a specific response pattern
   const startMockServer = (shouldSucceedOnSecondCall: boolean) => {
     // Create http server
-    server = http.createServer((req, res) => {
+    server = http.createServer({ keepAlive: false }, (req, res) => {
```

Comment:
Just curious, what happens when setting this to false?

Reply:
It simply ensures any outstanding http connections close once whatever you're doing on the spun up server concludes. It's sent as a default to `true` in Node20+, which is a new development. Unfortunately, setting it to `false` didn't fix the failing-tests-in-Node-20 problem :(